plugins/pulp_rpm/plugins/importers/yum/listener.py

```diff
@@ -78,30 +87,24 @@ def download_succeeded(self, report):
         # these are the only types we store repo metadata snippets on in the DB
         if isinstance(model, (models.RPM, models.SRPM)):
             self.metadata_files.add_repodata(model)
-        # init unit, which is idempotent
-        unit = self.sync_conduit.init_unit(model.TYPE, model.unit_key, model.metadata,
-                                           model.relative_path)
-        # check if the unit has duplicate nevra
-        repo_id = self.sync_conduit.repo_id
-        purge.remove_unit_duplicate_nevra(unit.unit_key, unit.type_id, repo_id)
-        # move to final location.
-        # we cannot use here shutil.move because it preserves all the file attributes,
-        # even the selinux labels from the the source directory, that has different label
-        # from the desination one.
-        # we dont't have to worry about the content from working directory as it gets cleaned up,
-        # when a task finishes.
-        shutil.copy(report.destination, unit.storage_path)
-        # save unit
-        self.sync_conduit.save_unit(unit)
+
+        purge.remove_unit_duplicate_nevra(model, self.sync_conduit.repo)
+
+        model.set_content(report.destination)
+        model.save()
+
+        repo_controller.associate_single_unit(self.sync_conduit.repo, model)
```

Comment:
I think after all units are downloaded, all unit counts are updated with [this call](https://github.com/pulp/pulp_rpm/pull/741/files#diff-5185b3f1e5a02c40a5d198fe779c75fbR268)

Aside from that we could probably replace some of the `rebuild_content_unit_counts()` to `update_unit_count()` in cases where a known number of units have been added or removed. +1 to doing a quick audit of that pre-merge.